pyLoadGui.py

```diff
@@ -186,80 +1124,253 @@ def slotShowConnector(self):
+        icon = "C"
+        btnSet = "OK"
+        tests()
+        icon = "N"
+        btnSet = "YES_NO"
+        tests()
+
+    def debugMsgBoxTest2(self):
+        def dm(n):
+            self.log.debug9("main.debugMsgBoxTest2: showing messageBox_%s" % n)
+        # Connector
+        host = "looooooooooooooooooooooooong.hostname.com"
+        port = 66666
+        server_version = '42'
+        dm("01"); self.connector.messageBox_01(host, port)
+        dm("02"); self.connector.messageBox_02(host, port)                          # no explicit parent
+        dm("03"); self.connector.messageBox_03(host, port, "BananaJoe", "tomato")   # no explicit parent, class AskForUserAndPassword
+        dm("04"); self.connector.messageBox_04(host, port)
+        dm("05"); self.connector.messageBox_05(host, port)
+        dm("06"); self.connector.messageBox_06(server_version, host, port)
+        # main
+        pidfile = "bob/pyload.pid"
+        pid = 536485
+        optCat = "FooBar"
+        dm("07"); self.messageBox_07()
+        dm("08"); self.messageBox_08()
+        dm("09"); self.messageBox_09()
+        dm("12"); self.messageBox_12()
+        dm("13"); self.messageBox_13()
+        dm("14"); self.messageBox_14()
+        dm("15"); self.messageBox_15((pidfile, pid))
+        dm("16"); self.messageBox_16()
+        dm("17"); self.messageBox_17()
+        dm("18"); self.messageBox_18(pid)
+        dm("21"); self.messageBox_21(optCat)
+        dm("22"); self.messageBox_22()
+        dm("23"); self.messageBox_23(pidfile, pid)
+        dm("24"); self.messageBox_24()
+        dm("25"); self.messageBox_25()
+        dm("26"); self.messageBox_26()
+        # ClickNLoadForwarder
+        dm("19"); self.clickNLoadForwarder.messageBox_19()
+        dm("20"); self.clickNLoadForwarder.messageBox_20()
+
+    def debugKill(self):
+        self.log.debug9("main.debugKill: terminate process")
+        self.quitConnTimeout()
+
+    def checkPackageAdded(self):
+        """
+            send a notification when a package was added to collector or queue
+        """
+        q = self.queue.getLastAddedPackage()            # (pid, name)
+        c = self.collector.getLastAddedPackage()        # (pid, name)
+        if q is None or c is None:
+            return
+
+        if q[0] == 0 and c[0] == 0:
+            # no packages, queue and collector are empty
+            if self.lastPackageId is None:  # initial poll
+                self.lastPackageId = 0      # valid package pids are > 0
+            return
+
+        if q[0] > c[0]:
+            (pid, name) = (q[0], q[1])
+        else:
+            (pid, name) = (c[0], c[1])
+        if self.lastPackageId is None:  # initial poll
+            self.lastPackageId = pid
+        else:
+            if pid > self.lastPackageId:
+                self.lastPackageId = pid
+                QTimer.singleShot(0, lambda: self.slotNotificationMessage(103, name))
+
+    def slotNotificationMessage(self, status, name):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [main.slotNotificationMessage is too complex (22)](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)